Attachments can be added on first publish

Rosa-Fox opened the pull request:
[Trello](https://trello.com/c/8jJTC2vu/155-adding-an-attachment-should-preserve-the-update-type-medium)
- Paired with @seraphiana
- Previously, if an attachment was added to a draft document, the document would fail to publish
- Ensures that update type is always major for draft document
- If document is not a draft then the user can select the preferred update_type
- republish_worker no longer sets a `republish` update_type for draft documents
- If the update_type for drafts was set to `republish` then emails wouldn't be sent. Content wouldn't be sent on the message queue if the document was subsequently published by an editor
- [v1 fix for similar issue](alphagov/specialist-publisher@8f5f504)

Testing - We felt this [exisiting test](https://github.com/alphagov/specialist-publisher-rebuild/blob/master/spec/features/editing_a_cma_case_spec.rb#L228) covered as much of the journey as we could achieve without full end to end testing

Rosa-Fox committed:
Attachments can be added on first publish

- Paired with @seraphiana
- Previously, if an attachment was added to a draft document, the document would fail to publish
- Ensures that update type is always major for draft document
- If document is not a draft then the user can select the preferred update_type

Testing - We felt this [exisiting test](https://github.com/alphagov/specialist-publisher-rebuild/blob/master/spec/features/editing_a_cma_case_spec.rb#L228) covered as much of the journey as we could achieve without full end to end testing

Seraphiana commented on app/models/document.rb:
```diff
@@ -121,7 +121,11 @@ def change_history
 
   def update_type
-    @update_type || "major"
+    if self.draft? && @update_type != "republish"
```
think the second half should no longer be here: even if a draft is republished it still needs to be a major update so that email alerts are sent

tuzz committed:
Don't send a 'republish' update_type for draft/redrafted documents

Notes from Rosa Fox:

- republish_worker no longer sets a `republish` update_type for draft
  documents
- If the update_type for drafts was set to `republish` then emails
  wouldn't be sent. Content wouldn't be sent on the message queue if
  the document was subsequently published by an editor
- [v1 fix for similar issue](https://github.com/alphagov/specialist-publisher/pull/728/commits/8f5f5046edd8a77aeb38575ebd566115c23a0db7)
